assets/js/components/notifications/notification.js

```diff
@@ -74,6 +89,19 @@ class Notification extends Component {
 		}, 350 );
 	}
 
+	handleCTAClick( e ) {
+		e.persist();
+
+		const { isDismissable, onCTAClick } = this.props;
+		const dismiss = isDismissable ? this.dismiss : () => {};
+
+		if ( onCTAClick ) {
+			onCTAClick( e, dismiss );
+		} else {
+			dismiss( e );
+		}
```

Comment:
Revisiting this, wouldn't it make more sense to make `handleCTAClick` `async` and then `await` `onCTAClick`, without passing `dismiss` to it? Then we would always call `dismiss` at the end. This would make it more similar to `handleDismiss` and avoid the extra parameter.

Reply:
Sure we could do that. My initial approach here assumes that there might be cases where we don't want to dismiss on CTA click, even if the notification is dismissible, but that does sound like a bit of an edge case 😄 

This looks cleaner indeed though so assuming the above isn't a concern, let's go with it 👍